### ISSUE TITLE
JSON::Serializable leads to a compilation error if included into Exception child

### ISSUE THREAD

**Issue by vladfaust** [crystal-lang/crystal#6998]
See https://carc.in/#/r/5d22, also see https://carc.in/#/r/5da3. I believe the issue can be narrowed, but I can't formulate it right.

**Comment by RX14** [comment]
The error says that `CallStack.new` doesn't accept a pullparser, which is because `CallStack` isn't serializable. `Exception` contains a `CallStack`. So this isn't a bug. You should be able to redefine `@callstack` with an ignore annotation.

**Comment by vladfaust** [comment]
Sorry, @RX14, could you please elaborate with a working example?

**Comment by z64** [comment]
@vladfaust https://carc.in/#/r/5da6

**Comment by vladfaust** [comment]
Awesome, @z64, thank you!

**Comment by RX14** [comment]
This should work but doesn't: https://carc.in/#/r/5daa

**Comment by vladfaust** [comment]
Yep, I was expecting including `JSON::Serializable` **only** into custom `Error` to work, because I never need to (de)serialize a plain `Exception`.

**Comment by RX14** [comment]
@asterite my example does that and still fails.

**Comment by asterite** [comment]
It seems putting an annotation over an inherited instance var doesn't work. I will try to take a look at it if I have time. 

Reduced:

```crystal
require "json"

class Foo
  @x : Int32?
end

class Bar < Foo
  @[JSON::Field(ignore: true)]
  @x : Int32?
end

def foo
  {% puts Foo.instance_vars.first.annotation(JSON::Field) %}
  {% puts Bar.instance_vars.first.annotation(JSON::Field) %}
end

foo
```

It prints `nil` in both cases but it should print the annotation in both cases.

(I would create a separate issue for this, JSON::Serializable is just a side-effect)

**Comment by asterite** [comment]
Or, well, just reopen `Exception` and mark `cause` as ignored, like one of the snippets above (please put the snippets here, it's hard to follow the conversation with just links pointing to play)

**Comment by asterite** [comment]
(so I won't take a look at this right now because it has a simple workaround)

**Comment by straight-shoota** [comment]
> it should print the annotation in both cases.

Really? I don't think adding an annotation in a subclass should add it in the parent.

**Comment by asterite** [comment]
There's only one instance var, inherited

**Comment by straight-shoota** [comment]
Hm, yeah true. It might still be unexpected to change the parent scope.
